File: Algorithms/922. Sort Array By Parity II/javascript/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParityII = function (nums) {
    let evenIdx = 0;
    let oddIdx = 1;

    while (evenIdx < nums.length && oddIdx < nums.length) {
        if (nums[evenIdx] % 2 === 0) {
            evenIdx += 2;
        } else if (nums[oddIdx] % 2 === 1) {
            oddIdx += 2;
        } else {
            let temp = nums[evenIdx];
            nums[evenIdx] = nums[oddIdx];
            nums[oddIdx] = temp;
            evenIdx += 2;
            oddIdx += 2;
        }
    }
    return nums;
};

File: Algorithms/922. Sort Array By Parity II/golang/solution.go
func sortArrayByParityII(nums []int) []int {
	length := len(nums)
	even_idx := 0
	odd_idx := 1

	for even_idx < length && odd_idx < length {
		if nums[even_idx]%2 == 0 {
			even_idx += 2
		} else if nums[odd_idx]%2 == 1 {
			odd_idx += 2
		} else {
			nums[even_idx], nums[odd_idx] = nums[odd_idx], nums[even_idx]
		}
	}
	return nums
}
